File: src/main/java/org/hotamachisubaru/miniutility/Nickname/NicknameCommand.java
package org.hotamachisubaru.miniutility.Nickname;

import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.NamedTextColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.hotamachisubaru.miniutility.Miniutility;

import java.sql.SQLException;

public class NicknameCommand implements CommandExecutor {
    private final Miniutility plugin;

    public NicknameCommand(Miniutility plugin) {
        this.plugin = plugin;
    }

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        if (!(sender instanceof Player)) {
            sender.sendMessage(Component.text("このコマンドはプレイヤーのみが使用できます。", NamedTextColor.RED));
            return true;
        }

        Player player = (Player) sender;

        if (args.length == 0) {
            player.sendMessage(Component.text("新しいニックネームを入力してください。", NamedTextColor.YELLOW));
            try {
                NicknameManager.setNickname(player, "");
            } catch (SQLException e) {
                throw new RuntimeException(e);
            }
            return true;
        }

        String nickname = String.join(" ", args).trim();
        try {
            NicknameManager.setNickname(player, nickname);
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
        player.displayName(Component.text(nickname));
        player.playerListName(Component.text(nickname));
        player.sendMessage(Component.text("ニックネームを " + nickname + " に設定しました！", NamedTextColor.GREEN));
        return true;
    }
}

File: src/main/java/org/hotamachisubaru/miniutility/Nickname/NicknameCommand.kt
package org.hotamachisubaru.miniutility.Nickname

import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import org.hotamachisubaru.miniutility.Miniutility

class NicknameCommand(private val plugin: Miniutility) : CommandExecutor {
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        if (sender !is Player) {
            sender.sendMessage("このコマンドはプレイヤーのみ実行できます。")
            return true
        }

        val player = sender
        val uuid = player.uniqueId
        val config = plugin.nicknameConfig

        if (args.size == 0) {
            // ニックネームのリセット
            config!!.setNickname(uuid, null)
            player.setDisplayName(player.name)
            player.setPlayerListName(player.name) // プレイヤーリスト名もリセット
            player.sendMessage(ChatColor.GREEN.toString() + "ニックネームをリセットしました。")
        } else {
            // ニックネームの設定
            val nickname = java.lang.String.join(" ", *args)
            config!!.setNickname(uuid, nickname)
            player.setDisplayName(nickname)
            player.setPlayerListName(nickname) // プレイヤーリスト名も更新
            player.sendMessage(ChatColor.GREEN.toString() + "ニックネームを " + nickname + " に設定しました。")
        }

        return true // チャットには影響を与えない
    }
}